src/integer_mod_q/mat_polynomial_ring_zq/get.rs

```diff
@@ -158,6 +158,54 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
 }
 
 impl MatPolynomialRingZq {
+    /// Returns a deep copy of the submatrix defined by the given parameters.
+    /// All entries starting from `(row1, col1)` to `(row2, col2)`(inclusively) are collected in
+    /// a new matrix.
+    /// Note that `row1 >= row2` and `col1 >= col2` must hold. Otherwise the function will panic.
+    ///
```

Comment:
Maybe implement get_row and get_column if it is not there.

Reply:
will make a separate PR for it